### PR TITLE
Vue3: Fix support for `onX` and empty attributes in Show Code

### LINKED ISSUE #25143: [Bug]: props containing 'on' are improperly rendered in "Show code"
### Describe the bug

In the "Show code" section, any prop containing "on" is rendered as a callback. The test should be whether the name starts with "on" rather than whether it contains "on".

### To Reproduce

https://stackblitz.com/edit/github-j6h4hd?file=src%2Fstories%2FButton.vue,src%2Fstories%2FButton.stories.js&preset=node

- Go to Button / Docs in the storybook
- Click "Show code"
- Edit the "icon" prop

-> The generated prop in "Show code" is `icon="()=>({})"` whatever the value you set.

### System

```bash
System:
    OS: Linux 6.2 Ubuntu 22.04.3 LTS 22.04.3 LTS (Jammy Jellyfish)
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i7-1165G7 @ 2.80GHz
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 20.10.0 - ~/.nvm/versions/node/v20.10.0/bin/node
    Yarn: 1.22.19 - ~/.yarn/bin/yarn <----- active
    npm: 10.2.3 - ~/.nvm/versions/node/v20.10.0/bin/npm
  Browsers:
    Chrome: 119.0.6045.199
  npmPackages:
    @storybook/addon-essentials: ^7.6.3 => 7.6.3 
    @storybook/addon-interactions: ^7.6.3 => 7.6.3 
    @storybook/addon-links: ^7.6.3 => 7.6.3 
    @storybook/blocks: ^7.6.3 => 7.6.3 
    @storybook/test: ^7.6.3 => 7.6.3 
    @storybook/vue3: ^7.6.3 => 7.6.3 
    @storybook/vue3-vite: ^7.6.3 => 7.6.3 
    storybook: ^7.6.3 => 7.6.3
```

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Tap-Kim**
Closes https://github.com/storybookjs/storybook/issues/25143

## What I did

We have added code supplementation and test code to correct the issues below.

```
Describe the bug
In the "Show code" section, any prop containing "on" is rendered as a callback. The test should be whether the name starts with "on" rather than whether it contains "on".

To Reproduce
https://stackblitz.com/edit/github-j6h4hd?file=src%2Fstories%2FButton.vue,src%2Fstories%2FButton.stories.js&preset=node

Go to Button / Docs in the storybook
Click "Show code"
Edit the "icon" prop
-> The generated prop in "Show code" is icon="()=>({})" whatever the value you set.
```
#### The changes in this PR are covered in the following automated tests:
- [ ] stories
- [x] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

     - `bug`: Internal changes that fixes incorrect behavior.
     - `maintenance`: User-facing maintenance tasks.
     - `dependencies`: Upgrading (sometimes downgrading) dependencies.
     - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
     - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
     - `documentation`: Documentation **only** changes. Will not show up in release changelog.
     - `feature request`: Introducing a new feature.
     - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
     - `other`: Changes that don't fit in the above categories.
   
   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

### CODE REVIEW

**Comment by JReinhold** [comment]
Thanks for the contribution!

Checking for "on" with `startsWith` instead of `includes` makes sense to me, but I don't fully understand the part about removing empty functions/objects.

If you're passing in a function as an arg, why shouldn't that be shown in the documentation, even if it's an empty function?

**Comment by Tap-Kim** [comment]
Hello @JReinhold
Could you check if there are 2 workflows waiting for approval for two weeks?

**Review by JReinhold — APPROVED** [review]

**Pull request merged**
